bindings/profiler.cc

```diff
@@ -101,61 +105,226 @@ CpuProfiler* cpuProfiler = CpuProfiler::New(v8::Isolate::GetCurrent());
+                    Nan::New<Integer>(parent->GetScriptId()),
+                    Nan::New<Integer>(node->GetLineNumber()),
+                    Nan::New<Integer>(node->GetColumnNumber()),
+                    Nan::New<Integer>(0),
+                    GetLineNumberTimeProfileChildren(parent, node));
+}
+
+// In profiles with line-level accurate line numbers, a node's line number
+// and column number refer to the line/column from which the function was
+// called.
+Local<Value> TranslateLineNumbersTimeProfileRoot(const CpuProfileNode* node) {
+  int32_t count = node->GetChildrenCount();
+  Local<Array> childrenArrs[count];
+  int32_t childCount = 0;
+  for (int32_t i = 0; i < count; i++) {
+    Local<Array> c = GetLineNumberTimeProfileChildren(node, node->GetChild(i));
+    childCount = childCount + c->Length();
+    childrenArrs[i] = c;
+  }
+
+  Local<Array> children = Nan::New<Array>(childCount);
+  int32_t idx = 0;
+  for (int32_t i = 0; i < count; i++) {
+    Local<Array> arr = childrenArrs[i];
+    for (int32_t j = 0; j < arr->Length(); j++) {
+      children->Set(idx, arr->Get(j));
+      idx++;
+    }
+  }
+
+  return CreateTimeNode(
+      node->GetFunctionName(),
+      node->GetScriptResourceName(),
+      Nan::New<Integer>(node->GetScriptId()),
+      Nan::New<Integer>(node->GetLineNumber()),
+      Nan::New<Integer>(node->GetColumnNumber()),
+      Nan::New<Integer>(0),
+      children
+  );
+}
+#endif
 
 Local<Value> TranslateTimeProfileNode(const CpuProfileNode* node) {
-  Local<Object> js_node = Nan::New<Object>();
-  js_node->Set(Nan::New<String>("name").ToLocalChecked(),
-    node->GetFunctionName());
-  js_node->Set(Nan::New<String>("scriptName").ToLocalChecked(),
-    node->GetScriptResourceName());
-  js_node->Set(Nan::New<String>("scriptId").ToLocalChecked(),
-    Nan::New<Integer>(node->GetScriptId()));
-  js_node->Set(Nan::New<String>("lineNumber").ToLocalChecked(),
-    Nan::New<Integer>(node->GetLineNumber()));
-  js_node->Set(Nan::New<String>("columnNumber").ToLocalChecked(),
-    Nan::New<Integer>(node->GetColumnNumber()));
-  js_node->Set(Nan::New<String>("hitCount").ToLocalChecked(),
-    Nan::New<Integer>(node->GetHitCount()));
   int32_t count = node->GetChildrenCount();
   Local<Array> children = Nan::New<Array>(count);
   for (int32_t i = 0; i < count; i++) {
     children->Set(i, TranslateTimeProfileNode(node->GetChild(i)));
   }
-  js_node->Set(Nan::New<String>("children").ToLocalChecked(),
-    children);
-  return js_node;
+
+  return CreateTimeNode(
+    node->GetFunctionName(),
+    node->GetScriptResourceName(),
+    Nan::New<Integer>(node->GetScriptId()),
+    Nan::New<Integer>(node->GetLineNumber()),
+    Nan::New<Integer>(node->GetColumnNumber()),
+    Nan::New<Integer>(node->GetHitCount()),
+    children
+  );
 }
 
-Local<Value> TranslateTimeProfile(const CpuProfile* profile) {
+Local<Value> TranslateTimeProfile(const CpuProfile* profile, bool hasDetailedLines) {
```

Comment:
"detailed" is an unclear term. Also "has" is not super clear - is this flag requesting something or stating something?

Reply:
Switched to using "includeLineInfo" (for starting the profile, when we're requesting line information), and "includedLineInfor" (where it indicates that line info is present).

#73